packages/core/src/tools/error.ts

```diff
@@ -110,3 +137,20 @@ export function createHandlingStack(): string {
 
   return formattedStack!
 }
+
+export function flattenErrorCauses(error: RawError, parentSource: ErrorSource): ErrorCause[] {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏this function should not take a `RawError` as argument. It should take an `Error` or `unknown` argument and check whether it has a `cause` property.